drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -510,102 +506,97 @@ static struct clock_control_driver_api stm32_clock_control_api = {
 	.get_rate = stm32_clock_control_get_subsys_rate,
 };
 
-static int stm32_clock_control_init(const struct device *dev)
+__unused
+static int set_up_pll3(void)
 {
-
-#if !defined(CONFIG_CPU_CORTEX_M4)
-	uint32_t old_hclk_freq = 0;
-	uint32_t new_hclk_freq = 0;
-
-#if STM32_PLL_SRC_HSE || \
-	STM32_PLL_SRC_HSI || \
-	STM32_PLL_SRC_CSI || \
-	STM32_PLL3_ENABLE
-
+#if defined(STM32_PLL3_ENABLE)
 	int r;
 	uint32_t vco_input_range;
 	uint32_t vco_output_range;
-#endif /* STM32_PLL_SRC_* */
-
-#endif /* ! CONFIG_CPU_CORTEX_M4 */
-
-	ARG_UNUSED(dev);
 
-#if !defined(CONFIG_CPU_CORTEX_M4)
+	/* Initialize PLL 3 */
+	r = get_vco_input_range(STM32_PLL3_M_DIVISOR, &vco_input_range);
+	if (r < 0) {
+		return r;
+	}
```

Comment:
Is the `stm32_clock_control_init` function returning 0 ?
In the clock_stm32_ll_common.c we read:
```
/**
 * @brief Initialize clocks for the stm32
 *
 * This routine is called to enable and configure the clocks and PLL
 * of the soc on the board. It depends on the board definition.
 * This function is called on the startup and also to restore the config
 * when exiting for low power mode.
 *
 * @param dev clock device struct
 *
 * @return 0
 */
```

Reply:
We'll probably need to fix this description, but I'll address this when reviewing this driver.